### PR TITLE
Initial Logs SOCKS5 Proxy Support

### PR DESCRIPTION
**Pull request by jstanton617**
### What does this PR do?

Allows the agent to connect to the logging ingestion endpoint via a SOCKS5-compatible proxy. Configure using:

```
logs_config:
  socks5_proxy_address: fqdn.example.com:port
```

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/logs/agent.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ func NewAgent(sources *config.LogSources) *Agent {
 		config.LogsAgent.GetString("logs_config.dd_url"),
 		config.LogsAgent.GetInt("logs_config.dd_port"),
 		config.LogsAgent.GetBool("logs_config.dev_mode_no_ssl"),
+		config.LogsAgent.GetString("logs_config.socks5"),
```
Comment (ajacquemot) [inline pkg/logs/agent.go]: Could you rename this in `logs_config.socks5_url` or `logs_config.socks5_address` or `logs_config.socks5_proxy_url` or `logs_config.socks5_proxy_address`  please ?

File: pkg/logs/sender/connection_manager.go [hunk 2]
```diff
@@ -60,7 +63,22 @@ func (cm *ConnectionManager) NewConnection() net.Conn {
 		}
 
 		cm.retries++
-		outConn, err := net.DialTimeout("tcp", cm.connectionString, timeout)
+
+		var outConn net.Conn
+		var err error
+
+		if cm.socksProxy != "" {
+			log.Info("Connecting to logs intake via socks5://", cm.socksProxy)
```
Comment (ajacquemot) [inline pkg/logs/sender/connection_manager.go]: I think this log should be moved to the `if cm.firstConn {...}` block.

**Review by ajacquemot — COMMENTED** [review]
I left two comments, LGTM overall, thanks for working on this.

**Comment by ajacquemot** [comment]
Can you run `inv test --targets=./pkg/logs/sender` to fix the linting issues please ?
You'll need to add a release note as well running `reno new socks5-proxy-support --edit` and sign the CLA.

File: pkg/logs/sender/connection_manager.go [hunk 3]
```diff
@@ -13,37 +13,36 @@ import (
...
-	mutex   sync.Mutex
-	retries int
-
-	firstConn bool
+	serverName    string
+	serverAddress string
+	devModeNoSSL  bool
+	proxyAddress  string
+	mutex         sync.Mutex
+	retries       int
```
Comment (tmichelet) [inline pkg/logs/sender/connection_manager.go]: do we still need `retries`?

File: pkg/logs/sender/connection_manager.go [hunk 4]
```diff
@@ -53,37 +52,55 @@ func (cm *ConnectionManager) NewConnection() net.Conn {
 	cm.mutex.Lock()
 	defer cm.mutex.Unlock()
 
+	var retries int
 	for {
 		if cm.firstConn {
-			log.Info("Connecting to the backend: ", cm.connectionString)
+			if cm.proxyAddress != "" {
```
Comment (tmichelet) [inline pkg/logs/sender/connection_manager.go]: could this `if cm.firstConn` be moved out of the for loop? It should happen once only

**Review by tmichelet — DISMISSED** [review]
nice, thanks for the cleanup! left two comments, feel free to merge once addressed.

**Review by tmichelet — DISMISSED** [review]

**Review by ajacquemot — APPROVED** [review]

**Pull request merged**